calico_node/startup/startup.go

```diff
@@ -65,32 +63,23 @@ func main() {
 	// An explicit value of true is required to wait for the datastore.
 	if os.Getenv("WAIT_FOR_DATASTORE") == "true" {
 		waitForConnection(client)
+		log.Info("Kubernetes is initialized as a Calico datastore")
 	} else {
 		message("Skipping datastore connection test")
 	}
 
-	// If this is a Kubernetes backed datastore then just make sure the
-	// datastore is initialized and then exit.  We don't need to explicitly
+	// If this is a Kubernetes backed datastore then we need to make sure the
+	// datastore is initialized.  We don't need to explicitly
 	// initialize the datastore for non-Kubernetes because the node resource
 	// management will do that for us.
 	if cfg.Spec.DatastoreType == api.Kubernetes {
 		message("Calico is using a Kubernetes datastore")
-		err = client.EnsureInitialized()
+		err := client.EnsureInitialized()
```

Comment:
If I remove this We end up getting an error related to IPPools:

```
heschlie@Dalaran:vagrant$ kubectl logs -n kube-system calico-node-w1xbv calico-node
Checking datastore connection
time="2017-03-17T21:56:05Z" level=info msg="Loading config from environment" 
time="2017-03-17T21:56:05Z" level=info msg="Datastore is ready" 
Datastore connection verified
Using autodetected IPv4 address on interface eth1: 172.18.18.102/24
No AS number configured on node resource, using global value
time="2017-03-17T21:56:05Z" level=info msg="Initialise BGP data" 
time="2017-03-17T21:56:05Z" level=info msg="Ensure default IPv4 pool is created. IPIP mode: off" 
ERROR: Failed to create default IPv4 IP pool: resource does not exist: IPPool(cidr=10.244.0.0/16)
Terminating
Calico node failed to start
```

But with the `EnsureInitialized()` called it starts successfully:

```
heschlie@Dalaran:vagrant$ kubectl logs -n kube-system calico-node-j1qqp calico-node
Checking datastore connection
time="2017-03-17T22:02:56Z" level=info msg="Loading config from environment" 
Datastore connection verified
Calico is using a Kubernetes datastore
time="2017-03-17T22:02:56Z" level=info msg="Datastore is ready" 
time="2017-03-17T22:02:56Z" level=info msg="Ensuring ThirdPartyResources exist" 
time="2017-03-17T22:02:56Z" level=info msg="Ensuring GlobalConfig ThirdPartyResource exists" 
time="2017-03-17T22:02:56Z" level=info msg="Ensuring IP Pool ThirdPartyResource exists" 
time="2017-03-17T22:02:56Z" level=info msg="ThirdPartyResources exist" 
time="2017-03-17T22:02:56Z" level=info msg="Ensuring ClusterType is set" 
time="2017-03-17T22:02:56Z" level=warning msg="Failed to apply ClusterType: resource does not exist: GlobalConfig(name=ClusterType)" 
time="2017-03-17T22:02:57Z" level=warning msg="Failed to apply ClusterType: resource does not exist: GlobalConfig(name=ClusterType)" 
time="2017-03-17T22:02:58Z" level=warning msg="Failed to apply ClusterType: resource does not exist: GlobalConfig(name=ClusterType)" 
time="2017-03-17T22:02:59Z" level=warning msg="Failed to apply ClusterType: resource does not exist: GlobalConfig(name=ClusterType)" 
time="2017-03-17T22:03:00Z" level=warning msg="Failed to apply ClusterType: resource does not exist: GlobalConfig(name=ClusterType)" 
time="2017-03-17T22:03:01Z" level=warning msg="Failed to apply ClusterType: resource does not exist: GlobalConfig(name=ClusterType)" 
Using autodetected IPv4 address on interface eth1: 172.18.18.101/24
No AS number configured on node resource, using global value
time="2017-03-17T22:03:02Z" level=info msg="ClusterType is set" 
time="2017-03-17T22:03:02Z" level=info msg="Using previously configured cluster GUID" 
time="2017-03-17T22:03:02Z" level=info msg="Kubernetes is initialized as a Calico datastore" 
time="2017-03-17T22:03:02Z" level=info msg="Initialise BGP data" 
time="2017-03-17T22:03:02Z" level=info msg="IPv6 support disabled through environment" 
time="2017-03-17T22:03:02Z" level=info msg="Ensuring ThirdPartyResources exist" 
time="2017-03-17T22:03:02Z" level=info msg="Ensuring GlobalConfig ThirdPartyResource exists" 
time="2017-03-17T22:03:02Z" level=info msg="Ensuring IP Pool ThirdPartyResource exists" 
time="2017-03-17T22:03:02Z" level=info msg="ThirdPartyResources exist" 
time="2017-03-17T22:03:02Z" level=info msg="Ensuring ClusterType is set" 
time="2017-03-17T22:03:02Z" level=info msg="ClusterType is set" 
time="2017-03-17T22:03:02Z" level=info msg="Using previously configured cluster GUID" 
Using node name: 172.18.18.101
time="2017-03-17T22:03:03Z" level=info msg="Loading config from environment" 
Calico node started successfully
snip...
```

I did notice that the `client.nodes.Apply()`, `client.nodes.Create()`, and `client.nodes.Update()` all seem to call `EnsureInitialized()` but `client.nodes.Get()` does not.